src/sentry/search/utils.py

```diff
@@ -468,6 +468,40 @@ def _run_latest_release_query(
     return [row[0] for row in cursor.fetchall()]
 
 
+def get_terminal_release_for_group(
+    group: Group,
+    query_type: LatestReleaseOrders,
+    latest: bool,
+) -> Sequence[str]:
+    direction = "DESC" if latest else "ASC"
+    extra_conditions = ""
+    if query_type == LatestReleaseOrders.SEMVER:
+        rank_order_by = f"major {direction}, minor {direction}, patch {direction}, revision {direction}, CASE WHEN (prerelease = '') THEN 1 ELSE 0 END {direction}, prerelease {direction}, sr.id {direction}"
+        extra_conditions += " AND sr.major IS NOT NULL"
+    else:
+        rank_order_by = f"COALESCE(date_released, date_added) {direction}"
+
+    query = f"""
+        SELECT sr.*
+        FROM sentry_release sr
+        INNER JOIN (
+            SELECT sgr.release_id
+            FROM sentry_grouprelease sgr
+            WHERE sgr.group_id = %s
+            ORDER BY sgr.release_id {direction}
+            -- We limit the number of groupreleases we check here to handle edge cases of groups with 100k+ releases
+            LIMIT 1000
+        ) sgr ON sr.id = sgr.release_id
+        {extra_conditions}
+        ORDER BY {rank_order_by}
```

Comment:
I'm concerned with the performance of this query if the columns are not indexed.
https://github.com/getsentry/sentry/blob/2fef7faed341554a5f1b5e26423ea1ef2212079f/src/sentry/models/release.py#L448

How can we determine if this is optimal? How can we measure it?

Reply:
The reason I do the subquery is to limit the cost of this outer query. So if we add the index on `group_id, release_id` then we should be ok. Doesn't matter how inefficient this outer sort is if we only have 1k rows.

That said, we do actually have an index for this semver sorting on the release table. I don't think it will be used here though, due to the subquery/join